src/vs/editor/contrib/inlineEdit/browser/inlineEditController.ts

```diff
@@ -0,0 +1,320 @@
+import { InlineEditHintsWidget } from 'vs/editor/contrib/inlineEdit/browser/inlineEditHintsWidget';
+import { EditorOption } from 'vs/editor/common/config/editorOptions';
+import { createStyleSheet2 } from 'vs/base/browser/dom';
+import { IConfigurationService } from 'vs/platform/configuration/common/configuration';
+
+export class InlineEditWidget implements IDisposable {
+	constructor(public readonly widget: GhostTextWidget, public readonly edit: IInlineEdit) { }
+
+	dispose(): void {
+		this.widget.dispose();
+	}
+}
+
+export class InlineEditController extends Disposable {
+	static ID = 'editor.contrib.inlineEditController';
+
+	public static readonly inlineEditVisibleKey = 'inlineEditVisible';
+	public static readonly inlineEditVisibleContext = new RawContextKey<boolean>(InlineEditController.inlineEditVisibleKey, false);
+	private _isVisibleContext = InlineEditController.inlineEditVisibleContext.bindTo(this.contextKeyService);
+
+	public static readonly cursorAtInlineEditKey = 'cursorAtInlineEdit';
+	public static readonly cursorAtInlineEditContext = new RawContextKey<boolean>(InlineEditController.cursorAtInlineEditKey, false);
+	private _isCursorAtInlineEditContext = InlineEditController.cursorAtInlineEditContext.bindTo(this.contextKeyService);
+
+	public static get(editor: ICodeEditor): InlineEditController | null {
+		return editor.getContribution<InlineEditController>(InlineEditController.ID);
+	}
+
+	private _currentEdit: ISettableObservable<InlineEditWidget | undefined> = this._register(disposableObservableValue(this, undefined));
+	private _currentRequestCts: CancellationTokenSource | undefined;
+
+	private _jumpBackPosition: Position | undefined;
+	private _isAccepting: boolean = false;
+
+	private readonly _enabled = observableFromEvent(this.editor.onDidChangeConfiguration, () => this.editor.getOption(EditorOption.inlineEdit).enabled);
+	private readonly _fontFamily = observableFromEvent(this.editor.onDidChangeConfiguration, () => this.editor.getOption(EditorOption.inlineEdit).fontFamily);
+	private readonly _backgroundColoring = observableFromEvent(this.editor.onDidChangeConfiguration, () => this.editor.getOption(EditorOption.inlineEdit).backgroundColoring);
+
+
+	constructor(
+		public readonly editor: ICodeEditor,
+		@IInstantiationService private readonly instantiationService: IInstantiationService,
+		@IContextKeyService private readonly contextKeyService: IContextKeyService,
+		@ILanguageFeaturesService private readonly languageFeaturesService: ILanguageFeaturesService,
+		@ICommandService private readonly _commandService: ICommandService,
+		@IConfigurationService private readonly _configurationService: IConfigurationService,
+	) {
+		super();
+
+		//Automatically request inline edit when the content was changed
+		//Cancel the previous request if there is one
+		//Remove the previous ghost text
+		this._register(editor.onDidChangeModelContent(async () => {
```

Comment:
Can you only subscribe to onDidChangeCursorPosition when enabled is true?

Reply:
✅ (Tho, check comment on `onDidChangeCursorPosition`)